src/oic/oic/message.py

```diff
@@ -674,6 +675,13 @@ class IdToken(OpenIDSchema):
     def verify(self, **kwargs):
         super(IdToken, self).verify(**kwargs)
 
+        try:
+            kwargs['iss'] == self['iss']
```

Comment:
Why should this produce an AssertionError? no assert anywhere...

It is also not a great idea to use 'assert' for this kind of code, as 'assert' gets removed when someone runs python with optimization (-O), so logic fails.